compiler/rustc_data_structures/src/lib.rs

```diff
@@ -131,6 +136,24 @@ impl<F: Fn()> Drop for OnDrop<F> {
     }
 }
 
+struct IterFromGenerator<G>(G);
+
+impl<G: Generator + Unpin> Iterator for IterFromGenerator<G> {
+    type Item = G::Yield;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match Pin::new(&mut self.0).resume(()) {
+            GeneratorState::Yielded(n) => Some(n),
+            GeneratorState::Complete(_) => None,
```

Comment:
Should we constrain `G::Return` to be `()` or would it be a burden?

Reply:
Done.